mobx/lib/src/core/context.dart

```diff
@@ -60,6 +60,7 @@ class ReactiveConfig {
     this.writePolicy = ReactiveWritePolicy.observed,
     this.readPolicy = ReactiveReadPolicy.never,
     this.maxIterations = 100,
+    this.isSpyEnabled = true,
```

Comment:
By default let's keep it false and have it be explicitly enabled.

Reply:
I've changed this piece as you described, but I'd like to note that the only way to enable spying now is by having a code that changes the `mainContext`'s config at runtime, like this:

```dart
if (needToSpy) {
  mainContext.config = ReactiveConfig.main.clone(isSpyEnabled: true); // or ReactiveConfig(isSpyEnabled: true)
  mainContext.spy(_handleMobxEvent);
}
```

instead of just having something like this:
```dart
if (needToSpy) {
  mainContext.spy(_handleMobxEvent);
}
```

The former looks acceptable to me, but I can't imagine a case where no-spying-by-default policy can be useful. Maybe you have some thoughts on this, @pavanpodila ?